packages/block-library/src/navigation/edit/placeholder/index.js

```diff
@@ -1,94 +1,117 @@
 /**
  * WordPress dependencies
  */
-import { Placeholder, Button } from '@wordpress/components';
+import { Placeholder, Button, Spinner } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { navigation, Icon } from '@wordpress/icons';
+import { speak } from '@wordpress/a11y';
+import { useEffect } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
-
 import useNavigationEntities from '../../use-navigation-entities';
 import PlaceholderPreview from './placeholder-preview';
 import useCreateNavigationMenu from '../use-create-navigation-menu';
 import NavigationMenuSelector from '../navigation-menu-selector';
 
 export default function NavigationPlaceholder( {
+	isSelected,
 	currentMenuId,
 	clientId,
-	onFinish,
-	hasResolvedNavigationMenus,
 	canUserCreateNavigationMenu = false,
+	isResolvingCanUserCreateNavigationMenu,
+	onFinish,
 } ) {
 	const createNavigationMenu = useCreateNavigationMenu( clientId );
 
 	const onFinishMenuCreation = async (
 		blocks,
 		navigationMenuTitle = null
 	) => {
-		if ( ! canUserCreateNavigationMenu ) {
-			return;
-		}
-
 		const navigationMenu = await createNavigationMenu(
 			navigationMenuTitle,
 			blocks
 		);
 		onFinish( navigationMenu, blocks );
 	};
 
-	const { isResolvingPages, isResolvingMenus } = useNavigationEntities();
-
-	const isStillLoading = isResolvingPages || isResolvingMenus;
+	const { isResolvingMenus, hasResolvedMenus } = useNavigationEntities();
 
 	const onCreateEmptyMenu = () => {
 		onFinishMenuCreation( [] );
 	};
 
+	useEffect( () => {
+		if ( ! isSelected ) {
+			return;
+		}
+
+		if ( isResolvingMenus ) {
+			speak(
+				__( 'Loading Navigation block setup placeholder options.' ),
```

Comment:
We can probably drop one of either 'setup' or 'placeholder' from this text and the text just below this. 

I think probably 'placeholder' can go because that's what we call it internally, but it's not something every user will understand (and it's often used in a different way).

```suggestion
				__( 'Loading Navigation block setup options.' ),
```

What do you think?

Reply:
Yeh `placeholder` makes no sense to users. It's a developer centric term 👍

Reply:
✅